src/textual/widgets/_data_table.py

```diff
@@ -2476,66 +2490,98 @@ def action_page_down(self) -> None:
+            row_index, _ = self.cursor_coordinate
             for ordered_row in self.ordered_rows[: row_index + 1]:
                 offset += ordered_row.height
+                rows_to_scroll += 1
                 if offset > height:
                     break
-                rows_to_scroll += 1
 
-            self.cursor_coordinate = Coordinate(
-                row_index - rows_to_scroll + 1, column_index
-            )
+            target_row = row_index - rows_to_scroll + 1
+            self.scroll_relative(y=-height, animate=False)
+            self.move_cursor(row=target_row, scroll=False)
         else:
             super().action_page_up()
 
-    def action_scroll_home(self) -> None:
-        """Scroll to the top of the data table."""
+    def action_page_left(self) -> None:
+        """Move the cursor one page left."""
+        self._set_hover_cursor(False)
+        super().scroll_page_left()
+
+    def action_page_right(self) -> None:
+        """Move the cursor one page right."""
+        self._set_hover_cursor(False)
+        super().scroll_page_right()
+
+    def action_scroll_top(self) -> None:
+        """Move the cursor and scroll to the top."""
         self._set_hover_cursor(False)
         cursor_type = self.cursor_type
         if self.show_cursor and (cursor_type == "cell" or cursor_type == "row"):
-            row_index, column_index = self.cursor_coordinate
+            _, column_index = self.cursor_coordinate
             self.cursor_coordinate = Coordinate(0, column_index)
         else:
             super().action_scroll_home()
 
-    def action_scroll_end(self) -> None:
-        """Scroll to the bottom of the data table."""
+    def action_scroll_bottom(self) -> None:
+        """Move the cursor and scroll to the bottom."""
         self._set_hover_cursor(False)
         cursor_type = self.cursor_type
         if self.show_cursor and (cursor_type == "cell" or cursor_type == "row"):
-            row_index, column_index = self.cursor_coordinate
+            _, column_index = self.cursor_coordinate
             self.cursor_coordinate = Coordinate(self.row_count - 1, column_index)
         else:
             super().action_scroll_end()
 
+    def action_scroll_home(self) -> None:
+        """Move the cursor and scroll to the leftmost column."""
+        self._set_hover_cursor(False)
+        cursor_type = self.cursor_type
+        if self.show_cursor and (cursor_type == "cell" or cursor_type == "column"):
+            self.move_cursor(column=0)
+        else:
+            self.scroll_x = 0
+
+    def action_scroll_end(self) -> None:
+        """Move the cursor and scroll to the rightmost column."""
+        self._set_hover_cursor(False)
+        cursor_type = self.cursor_type
+        if self.show_cursor and (cursor_type == "cell" or cursor_type == "column"):
+            self.move_cursor(column=len(self.columns) - 1)
+        else:
+            self.scroll_x = self.max_scroll_x
```

Comment:
Quickly jumping to the first/last cell in a row. Also intended to improve horizontal navigation with the keyboard.